src/main/java/dev/morling/onebrc/CalculateAverage_justplainlaake.java

```diff
@@ -0,0 +1,450 @@
+                // Also Math.min doesn't matter here since its called x times where x = count of processors
+
+                if (i == processors - 1) {// if on last processor use main thread to optimize threading, doing on last processor means the others are already processing while this runs
+                    mainMap = process(finalChunkStart, finalChunkEnd);
+                }
+                else {
+                    futures.add(e.submit(() -> process(finalChunkStart, finalChunkEnd)));
+                }
+                chunkStart = currentAddress + 1;// Advance the start of the next chunk to be the end of this chunk + 1 to move past the new line character
+                currentAddress = Math.min(currentAddress + chunkSize, endAddress);// Advance the next chunks end to be the end of the mapped file or the end of the approximated chunk
+            }
+        }
+
+        OpenMap merged = mainMap;// Set the main map created with the process called on main thread to make it effectively final
+
+        // The merging of processing takes ~10ms
+        for (Future<OpenMap> f : futures) {
+            try {
+
+                OpenMap processed = f.get();// Waits until the process task is done but then returns the callable value from the process method
+
+                // Simple way to merge both lists, tried doing it more inline inside the map and ended up taking a 10ms longer
+                processed.forEach((i, s) -> {
+                    merged.merge(i, s);
+                });
+            }
+            catch (InterruptedException | ExecutionException e1) {
+                e1.printStackTrace();
+            }
+        }
+
+        // Mark threadpool to be shutdown, call it here to let the threadpool finish out while the rest of the processing occurs
+        e.shutdown();
+
+        // Ordering and printing takes 50ms
+        Station[] nameOrdered = merged.toArray();// Turn the merged map into an array to quickly sort it
+
+        Arrays.sort(nameOrdered, (n1, n2) -> n1.name.compareTo(n2.name));// Sort based on name, this might be optimizable based on the longs of the name, but would likely only gain some ms??
+
+        // Print results to the sys out
+        System.out.print("{");
+        for (int i = 0; i < nameOrdered.length; i++) {
+            if (i != 0) {
+                System.out.print(", ");
+            }
+            System.out.print(nameOrdered[i]);
+        }
+        System.out.print("}\n");// Need newline character to meet specs
+    }
+
+    // Core processing functionality, processes a chunk of memory
+    private static OpenMap process(long fromAddress, long toAddress) {
+
+        OpenMap stationsLookup = new OpenMap();// Create a new map for this specific chunk, this is also the returned value for the callable
+
+        long blockStart = fromAddress;
+        long currentAddress = fromAddress;
+
+        while (currentAddress < toAddress) {// Just keep looping until we exhaust the chunk
+
```

Comment:
Something seems off with the chunking logic. When I print those two variables, I get this:

```
current: xxxxxxxxxxxx2832
     to: xxxxxxxxxxxx2836
```

So it only process the first line from that dot file but not the lines for "-".

Reply:
Interesting... weird that it chunked fine in my local 🤔 

Anyway, added a bypass in the chunking logic that I was thinking of adding anyway to improve performance on small files since the multithreading added overhead that was unneeded for small files. This will also fix this edge case that you have shown.

Tested again in my local after running `mvn clean package` 
```
Finished generating 'CalculateAverage_justplainlaake_image' in 32.0s.
Validating calculate_average_justplainlaake.sh -- src/test/resources/samples/measurements-1.txt
Validating calculate_average_justplainlaake.sh -- src/test/resources/samples/measurements-10.txt
Validating calculate_average_justplainlaake.sh -- src/test/resources/samples/measurements-10000-unique-keys.txt
Validating calculate_average_justplainlaake.sh -- src/test/resources/samples/measurements-2.txt
Validating calculate_average_justplainlaake.sh -- src/test/resources/samples/measurements-20.txt
Validating calculate_average_justplainlaake.sh -- src/test/resources/samples/measurements-3.txt
Validating calculate_average_justplainlaake.sh -- src/test/resources/samples/measurements-boundaries.txt
Validating calculate_average_justplainlaake.sh -- src/test/resources/samples/measurements-complex-utf8.txt
Validating calculate_average_justplainlaake.sh -- src/test/resources/samples/measurements-dot.txt
Validating calculate_average_justplainlaake.sh -- src/test/resources/samples/measurements-rounding.txt
Validating calculate_average_justplainlaake.sh -- src/test/resources/samples/measurements-short.txt
Validating calculate_average_justplainlaake.sh -- src/test/resources/samples/measurements-shortest.txt
[laake@DESKTOP-V6M7L3U:/mnt/c/Users/Laake/vscode-workspace/1brc]$ 
```